plugins/pulp_rpm/plugins/importers/yum/upload.py

```diff
@@ -512,6 +515,36 @@ def _extract_drpm_data(drpm_filename):
     return _encode_as_utf8(drpm_data)
 
 
+def _handle_modules(repo, type_id, unit_key, metadata, file_path, conduit, config):
+    """
+    Handles the upload of a modules.yaml file
+
+    :param repo: The repository to import the module data into
+    :type  repo: pulp.server.db.model.Repository
+
+    :param type_id: The type_id of the modulemd or modulemd-default being uploaded
+    :type  type_id: str
+
+    :param unit_key: A dictionary of fields to overwrite introspected field values
+    :type  unit_key: dict
+
+    :param metadata: A dictionary of fields to overwrite introspected field values, or None
+    :type  metadata: dict or None
+
+    :param file_path: The path to the uploaded modules.yaml file
+    :type  file_path: str
+
+    :param conduit: provides access to relevant Pulp functionality
+    :type  conduit: pulp.plugins.conduits.upload.UploadConduit
+
+    :param config: plugin configuration for the repository
+    :type  config: pulp.plugins.config.PluginCallConfiguration
+    """
+    modulemds, defaults = modules.from_file(file_path)
```

Comment:
in the description of the story we mention that we accept a compressed yaml doc, does it make sense to accept the yaml as it is or compressed format?

Reply:
Looks like I missed that. Sorry. I can definitely update this PR to have the upload accept compressed yaml. I don't have any strong opinions one way or the other but it looks like we accept an uncompressed comps.xml file.

Reply:
Today during the rpm meeting we talked about supporting both and checking based on the filename extension. That sounded like the best solution.